src/network/network_admin.cpp

```diff
@@ -333,11 +332,27 @@ NetworkRecvStatus ServerNetworkAdminSocketHandler::SendCompanyInfo(const Company
 	p->Send_uint8 (CeilDiv(c->months_of_bankruptcy, 3)); // send as quarters_of_bankruptcy
 
 	for (auto owner : c->share_owners) {
-		p->Send_uint8(owner);
+	  p->Send_uint8(owner);
+	}
+
+	std::vector<Company *> companies;
+	for(Company* company: Company::Iterate()){
+	  companies.emplace_back(company);
+	}
+
+	std::sort(companies.begin(), companies.end(), [](Company * const &c1, Company * const &c2){
+	  return c2->old_economy[0].performance_history < c1->old_economy[0].performance_history;
+	});
+
+	for(size_t t = 0; t < companies.size(); t++){
+	  companies[t]->rating = t+1;
```

Comment:
I am wondering about this block of logic, if that is the right thing to do. We manually sort the performance history here, and deduce a rating out of that. But why not let the admin-port clients do that? And just send over the `performance_history` to them? Saves us a lot of complex calculations, gives them more information than they receive now, and avoids nasty problems like: if you now request two companies with a slight delay, they might have the exact same rating as they so happened to have changed `performance_history`.

If we would go for this `rating` solution, the logic here could use some cleverness. You now calculate the rating for all companies, while you are only interested in a single one. So why not just calculate the single one? For example, it could be something like this (this replaces line 338 till 349 in your code):

```
int rating = 1;
for (auto company: Company::Iterate()) {
  if (c->old_economy[0].performance_history < company->old_economy[0].performance_history) rating++;
}
```

Basically, we find all companies that have a higher rating than us, meaning we drop one rating. The result is the exact same, but without the two loops and the sorting :) At least .. I think it is the same :P My simple testing shows it is :D But you do need to test it first :)

Reply:
Yeah that algorithm is simpler, the current way of getting everything, sorting it, and then counting, is excessive.
It could also be written with a `std::count_if()` call.